ftdc/custom_format.go

```diff
@@ -188,30 +197,35 @@ func flattenStruct(item reflect.Value) ([]float32, error) {
 // The function right now does not recursively walk data structures. We assume for now that the
 // caller will only feed "already flat" structures into FTDC. Later commits will better validate
 // input and remove limitations.
-func getFieldsForStruct(item reflect.Type) ([]string, error) {
-	flattenPtr := func(inp reflect.Type) reflect.Type {
-		for inp.Kind() == reflect.Pointer {
+func getFieldsForStruct(item reflect.Value) ([]string, error) {
```

Comment:
> I'm also concerned (FUD until proven otherwise) that merging them will make the "steady state" path of just caring about float32 values slower.

I don't understand this piece, but not merging for now despite the common `reflect.Value` param seems fine to me.

Reply:
I'm assuming you're curious why I think it would be less performant.

I think to merge the functions we'd need something like this:
```
type NameValue struct{
  name string
  value float32
}

func getFieldNamesAndValuesForStruct(item reflect.Value) []NameValue {
...
}
```

And now when I get the schema, I'm creating a bunch of strings (necessary) and a bunch of floats (unnecessary). That part I'm not too worried about. I worry more about the "steady state" where the schema doesn't change and we're just recording new values.

In that case we create a bunch of strings (unnecessary) and a bunch of floats (necessary). I worry that all of the extra allocations, just to be thrown away will "quickly" add up.

Reply:
Ah that makes sense to me; didn't immediately understand "steady state". Thanks for more context. Again, leaving as separate for now seems good.